rosetta/retriever/retriever_test.go

```diff
@@ -531,15 +581,22 @@ func TestRetriever_Block(t *testing.T) {
 		}
 
 		r := &Retriever{
+			cfg:       Config{TransactionLimit: 0}, // No transactions in block, all returned separately.
```

Comment:
A limit of zero is an edge case, remember boundary conditions? In this case, we would have the following boundaries:

- `TransactionLimit == 0`
- `TransactionLimit > 0 && TransactionLimit < numTransactions`
- `TransactionLimit == numTransactions`

I think it would be safe to discard the case where it is zero, because it will probably not be used, but we definitely need a test case where some transactions go into the block, and some transactions go into other transactions.